PsfLauncher/README.md

```diff
@@ -0,0 +1,99 @@
+</Package>
+```
+
+A possible `config.json` example would be:
+
+
+```json
+{
+	"applications": [
+	{
+		"id": "PSFLAUNCHERSixFour",
+      		"executable": "PrimaryApp.exe",
+      		"workingDirectory": "",
+      		"monitor": {
+			"executable": "PsfMonitor.exe",
+			"arguments": "",
+			"asadmin": true,
+        		"wait": false
+      		}
+    	}
+  	],
+	"processes": [
+	{
+		"executable": ".*PrimaryApp.*",
+		"shims": [ 
+		{ 
+			"dll": "TraceFixup64.dll",
+			"config": {
+					"traceMethod": "eventlog",
+					"traceLevels": {
+						"default": "always"
+					}
+			}
+		} 
+	        ]
+    	}
+  	]
+}
+```
+
+In this example, the configuration is directing the PsfLauncher to start PsfMonitor and then the referenced Primary App. PsfMonitor to be run using RunAs, followed by PrimaryApp once the monitoring app is stable. The root folder of the package is for the CurrentDirectory.  In the processes section of the example, the json further configures Trace Fixup to be injected into the Primary app, and that is to capture all levels of trace to the event log.
+
+
+| Array | key | Value |
+|-------|-----------|-------|
+| applications | id |  Use the value of the `Id` attribute of the `Application` element in the package manifest. |
+| applications | executable | The package-relative path to the executable that you want to start. In most cases, you can get this value from your package manifest file before you modify it. It's the value of the `Executable` attribute of the `Application` element. |
+| applications | workingDirectory | (Optional) A package-relative path to use as the working directory of the application that starts. If you don't set this value, the operating system uses the `System32` directory as the application's working directory. If you supply a value in the form of an empty string, it will use the directory of the referenced executable. |
+| applications | monitor | (Optional) If present, the monitor identifies a secondary program that is to be launched prior to starting the primary application.  A good example might be `PsfMonitor.exe`.  The monitor configuration consists of the following items:
+	`'executable'` - This is the name of the executable relative to the root of the package.
+	`'arguments'`  - This is a string containing any command line arguments that the monitor executable requires.
+	`'asadmin'` - This is a boolean (0 or 1) indicating if the executable needs to be launched as an admin.  To use this option you must mark the package with the RunAsAdministrator capability.  If the monitor executable has a manifest (internal or external) it is ignored.  If not expressed, this defaults to a 0.
+	`'wait'` - This is a boolean (0 or 1) indicating if the launcher should wait for the monitor program to exit prior to starting the primary application.  When not set, the launcher will WaitForInputIdle on the monitor before launching the primary application. This option is not normally used and defaults to 0. |

```

Comment:
Is there a scenario that it will be true?

Reply:
wait = true would not be appropriate for PsfMonitor.  In the past I have been hired by enterprises to write software that might run in PsfMonitor's place that does require wait=true.  This might be something as simple as a program to ensure a file share is mapped to a particular drive letter, to "signaling software" added by the customer to ensure that they are license compliance with a vendor's product that doesn't have an enforcement mechanism..

Reply:
Same as above, if there is no use for it today, lets remove it.
Also, on the "asadmin" option - how I decide when to turn it on? is it depends on weather the actual app proccess runs as admin?

Reply:
Please remove the properties that are not in use; if “wait” is not appropriate lets remove it, same for arguments, I prefer not to ship options that don’t have real usage.
Also, regard the asadmin option – would the psf Monitor behave differently when it is 0 vs 1?